ZenPacks/zenoss/Microsoft/Windows/txwinrm_utils.py

```diff
@@ -56,96 +59,102 @@ def createConnectionInfo(device_proxy):
     found to be invalid.
 
     """
-    if not hasattr(device_proxy, 'windows_servername'):
-        raise UnauthorizedError(
-            "attempted Windows connection to non-Windows device")
-
-    hostname = device_proxy.windows_servername or device_proxy.manageIp
-
-    username = device_proxy.windows_user
+    def getProxyValue(props):
+        if not isinstance(props, list):
+            props = [props]
+        for p in props:
+            if hasattr(device_proxy, p):
+                attr = getattr(device_proxy, p)
+                val = attr() if callable(attr) else attr
+                if val:
+                    return val
+        return ""
+
+    hostname = getProxyValue(['windows_servername', 'zWinRMServerName', 'manageIp'])
+    if not hostname:
+        raise UnauthorizedError("Attempted Windows connection to non-Windows device")
+
+    hostname = getProxyValue(['windows_servername', 'zWinRMServerName', 'manageIp'])
+
+    username = getProxyValue(['windows_user', 'zWinRMUser', 'zWinUser'])
     if not username:
-        raise UnauthorizedError("zWinRMUser or zWinUser must be configured")
+        raise UnauthorizedError(
+            "Cannot connect to Windows with an unknown user, zWinRMUser or zWinUser must be configured")
 
     # Warn about old-style usernames of the DOMAIN\User format.
-    pattern = r'[a-zA-Z0-9][a-zA-Z0-9.]{0,14}\\[^"/\\\[\]:;|=,+*?<>]{1,104}'
-    if re.match(pattern, username):
-        raise UnauthorizedError(
-            "zWinRMUser must be user@example.com, not DOMAIN\User")
+    if re.match(r'[a-zA-Z0-9][a-zA-Z0-9.]{0,14}\\[^"/\\\[\]:;|=,+*?<>]{1,104}', username):
+        raise UnauthorizedError("zWinRMUser must be user@example.com, not DOMAIN\User")
 
-    password = device_proxy.windows_password
+    password = getProxyValue(['windows_password', 'zWinRMPassword'])
     if not password:
-        raise UnauthorizedError(
-            "zWinRMPassword or zWinPassword must be configured")
+        raise UnauthorizedError("zWinRMPassword or zWinPassword must be configured")
```

Comment:
The message mentions "zWinPassword", but that zprop isn't searched for in the preceeding getProxyValue() call.

Reply:
You're right, zWinPassword was taken out many versions ago.  However, the message was intentionally kept this way (for reasons lost to time.)  I don't see why it needs to be here anymore, removing.